geospaas/base_viewer/views.py

```diff
@@ -56,55 +43,42 @@ def post(self, request, *args, **kwargs):
             self.forms[counter] = element_form_class(request.POST)
 
         self.validation_and_error_check()
-        # modify attributes based on self.form
-        return self.sorting_func(request)
+
+        # modify attributes based on the forms cleaned data
+        self.filtering_func(request)
+        return self.final_rendering(request)
```

Comment:
final_rendering should not call set_context which calls set_dataset_context. "Flat is better than nested"
Post should call all:
```suggestion
        ds = self.get_all_datasets()
        ds = self.filter_datasets(ds)
        contexts = self.set_context(ds)
        return render(request, self.main_template, context)
        
```

Reply:
The same for get()

Reply:
done

Reply:
for both get and post functions